### ISSUE TITLE
Bug: DLM TCS Distillation has critical logit misalignment causing 50+ loss

### ISSUE THREAD

**Issue by Lev-Stambler** [DeepOpt-com/WrinkleFree-Monorepo#13]
## Summary

TCS (Target Concrete Score) distillation for DLM students has a critical bug causing extremely high loss (~50-56) that doesn't decrease. Root cause identified by Gemini CLI analysis.

### Critical Misalignment in `trainer.py`

**Current (Bugged) Code:**
```python
# We assumed DLM predicts CURRENT token (like BERT)
aligned_teacher_logits = teacher_logits[:, :-1, :].contiguous()
aligned_student_logits = student_logits[:, 1:, :].contiguous()  # WRONG!
```

**Problem:**
- The student model is loaded via `AutoModelForCausalLM` → it's a **Causal LM that predicts NEXT token**
- We incorrectly assumed DLM predicts current token (like BERT/masked LM)
- Current alignment: Teacher[i] predicts x_{i+1}, but Student[i+1] predicts x_{i+2}
- We're training the student to output x_{i+2} when teacher says x_{i+1} → **WRONG TARGET**

**Correct Alignment:**
```python
# Both are CausalLMs - no shifting needed for student
aligned_teacher_logits = teacher_logits[:, :-1, :].contiguous()
aligned_student_logits = student_logits[:, :-1, :].contiguous()  # Same as teacher!
```

### Loss Scaling Issue

The loss is further inflated by aggressive scaling:
- `lambda_tcs = 10.0`
- `temperature = 5.0` → T² = 25.0
- **Effective weight: 10 × 25 = 250x**

If KL divergence is ~0.2 (typical), then 0.2 × 250 = 50, plus CE (~4) = **54 total loss**

## Fundamental Question

There's a deeper architectural question: **Is our DLM really a DLM?**

The checkpoint is loaded as `BitNetForCausalLM` via `AutoModelForCausalLM`. This means:
- Input: [a, [MASK], c, d, ...]
- Output at position 1: predicts token AFTER [MASK] (i.e., "c"), NOT the masked token "b"

For true DLM (masked prediction), we might need:
1. A different model architecture (masked LM head)
2. Different loss computation (predict at masked positions only)
3. Or accept that we're doing **causal distillation with corrupted input** (not true DLM)

## Files Affected

- `packages/distillation/src/distillation/training/trainer.py` - alignment logic (lines ~506-537)
- `packages/distillation/src/distillation/losses/tcs_loss.py` - loss computation
- `packages/distillation/configs/distillation/tcs.yaml` - lambda_tcs value

### Fix 1: Correct Alignment (if keeping masking)
```python
# In trainer.py _forward_step
aligned_teacher_logits = teacher_logits[:, :-1, :].contiguous()
aligned_student_logits = student_logits[:, :-1, :].contiguous()  # CHANGED
aligned_labels = labels[:, 1:].contiguous()

# Attention alignment
aligned_student_attentions = tuple(a[:, :, :-1, :-1].contiguous() for a in student_attentions)  # CHANGED
aligned_teacher_attentions = tuple(a[:, :, :-1, :-1].contiguous() for a in teacher_attentions)
```

### Fix 2: Reduce Loss Scale
```yaml
# In tcs.yaml
lambda_tcs: 1.0  # Reduced from 10.0
```

### Fix 3: Reconsider Masking Strategy
If the model is truly CausalLM, masking 15% of input tokens may not make sense. Consider:
- Remove masking entirely (standard AR→AR distillation)
- Or implement proper block-diffusion inference if the model supports it

## Environment

- Student checkpoint: `gs://wrinklefree-checkpoints/dlm/bitnet-b1.58-2B-4T-bf16/checkpoint-step-3400/`
- Teacher model: `microsoft/bitnet-b1.58-2B-4T-bf16`
- Loss observed: 50-56 (should be ~3-5 for working distillation)

## Related

- Plan file: `~/.claude/plans/vivid-plotting-sun.md`
- Branch: `GetDistillWorking`
